test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/Validation/DefaultObjectValidatorTests.cs

```diff
@@ -653,16 +663,14 @@ public void DictionaryType_ValidationSuccessful()
 
             // Assert
             Assert.True(validationContext.ModelState.IsValid);
-            var modelState = validationContext.ModelState["items"];
-            Assert.Equal(modelState.ValidationState, ModelValidationState.Valid);
-            modelState = validationContext.ModelState["items[0].Key"];
-            Assert.Equal(modelState.ValidationState, ModelValidationState.Skipped);
+            var modelState = validationContext.ModelState["items[0].Key"];
```

Comment:
Should assert the `Count` at least a few places in this class.  Only the tests using `Assert.Single()` seem to do so now.